concepts/solution-design-kafka-pull.md

```diff
@@ -0,0 +1,248 @@
+which provides a step-by-step message flow.
+
+### A. Provisioning/Deprovisioning Phase
+
+**Purpose:** Securely create or delete consumer credentials.
+
+**Flow:**
+
+1. **Provisioning:**
+    - The **Consumer Control Plane** sends a **TransferRequestMessage** to the **Provider Control Plane**.
+    - The **Provider Control Plane** forwards a **ProvisionRequest** to the **Kafka Extension**.
+    - The **Kafka Extension** calls the **OAuth Service** to create new credentials.
+    - Once credentials are issued, the Kafka Extension saves them securely in the Vault.
+    - A **ProvisionResponse** is then returned from the Kafka Extension up through the Provider Control Plane back to
+      the Consumer.
+2. **Deprovisioning:**
+    - When the transfer concludes, the **Provider Control Plane** issues a **DeprovisionRequest** to the Kafka
+      Extension.
+    - The Kafka Extension instructs the **OAuth Service** to delete the credentials.
+    - After credential deletion, the Kafka Extension clears the corresponding entries from the Vault and returns a *
+      *DeprovisionResponse**.
+
+[Sequence diagram of provisioning/deprovisioning using Kafka Extension](/concepts/MVP-kafka-pull/puml/Sequence%20diagram%20EDC%20Kafka%20Extension%20provisioning-deprovisioning.puml):
+
+![Sequence diagram EDC Kafka Extension provisioning-deprovisioning.png](/concepts/MVP-kafka-pull/png/Sequence%20diagram%20EDC%20Kafka%20Extension%20provisioning-deprovisioning.png)
+
+### B. Initiating the Transfer
+
+**Purpose:** Initiate a transfer process with dynamic credentials and EDR creation.
+
+**Flow:**
+
+1. **Start Transfer Request:**
+    - The **Consumer Control Plane** instructs the **Provider Control Plane** to start the transfer process.
+    - The Provider Control Plane performs policy and contract verification.
+2. **DataFlowRequest:**
+    - The Provider Control Plane sends a **DataFlowRequest** to the Kafka Extension.
+    - The Kafka Extension retrieves credentials from the Vault.
+    - It then communicates with the **Data Plane** to initiate the transfer.
+    - The Data Plane contacts the **OAuth Service** to generate both an access token and a refresh token.
+    - After token generation, the Data Plane creates a **DataAddress** containing necessary connection details (such as
+      topic, poll duration, OAuth token, and groupPrefix).
+    - A **DataFlowResponse** message is returned to the Kafka Extension, which adjusts the DataAddress if needed and
+      passes it back to the Provider Control Plane.
+    - Finally, the Provider Control Plane sends a **TransferStartMessage** (attached with the complete DataAddress) to
+      the Consumer Control Plane.
+    - The Consumer Control Plane then constructs the final Endpoint Data Reference (EDR) for use by the consumer
+      application.
+
+[Sequence diagram of start transfer process](/concepts/MVP-kafka-pull/puml/Sequence%20diagram%20EDC%20Kafka%20Extension%20start%20transfer%20process.puml):
```

Comment:
responses lines should be dotted

Reply:
They are where the flow is synchronous. The `Send TransferStartMessage attaching DataAddress` is an asynchronous message sent by the provider control plane to the consumer control plane